src/base/bittorrent/session.cpp

```diff
@@ -1218,10 +1223,7 @@ bool Session::addTorrent(const TorrentInfo &torrentInfo, const AddTorrentParams
 bool Session::addTorrent_impl(AddTorrentData addData, const MagnetUri &magnetUri,
                               const TorrentInfo &torrentInfo, const QByteArray &fastresumeData)
 {
-    if (!addData.resumed) {
-        if (addData.savePath.isEmpty() && isASMDisabledByDefault())
-            addData.savePath = m_defaultSavePath;
-    }
+    addData.savePath = normalizeSavePath(addData.savePath, (!addData.resumed ? m_defaultSavePath : ""));
```

Comment:
>Are you sure you don't want to check for isASMDisabledByDefault() too?

Oops!.. I want, of course. But the problem here is different. What you described is incorrect. An empty path in resumeData only means that the torrent is using Advanced saving mode.
Fixed.